src/consensus/aft/raft.h

```diff
@@ -1195,16 +1194,6 @@ namespace aft
           state->commit_idx);
         return;
       }
-      else if (r.prev_idx > state->last_idx)
-      {
-        RAFT_DEBUG_FMT(
-          "Recv append entries to {} from {} but prev_idx ({}) > last_idx ({})",
-          state->node_id,
-          from,
-          r.prev_idx,
-          state->last_idx);
-        return;
-      }
```

Comment:
Nit. We could replace it with some sort of assertion (or tracing) if we truly believe this cannot happen.
Cons
- one (possibly) redundant check
Pros
- we'll know if this happens

Feel free to close without changing.